lua/ccc/utils/pattern.lua

```diff
@@ -34,6 +34,11 @@ end
 ---@return integer? end
 ---@return string? ... submatches
 function pattern.find(str, pat, init)
+  -- Avoid vim:E976 error when the given string has a blob.
+  if vim.fn.type(str) ~= vim.v.t_string then
+    return
+  end
```

Comment:
A more rigorous form, avoiding calls to vim functions, is preferred.
```lua
-- matchlist() considers a string containing `\n` as a blob and cannot process them.
if str:find("\n") then
  return
end
```

Reply:
@uga-rosa unfortunately this change doesn't solve the issue (see my comment [here](https://github.com/uga-rosa/ccc.nvim/issues/101#issuecomment-1977910652)).

If using the code I originally proposed, then the issue _is_ truly fixed.

Reply:
Oh, sorry. This was a misunderstanding. It's a null character, not an `\n`.

Reply:
fixed in 86299a4562c86c29295361721f7229a0b99a17b8.

Reply:
Thank you!